### PR TITLE
Add styling customizations to mwc-button

### LINKED ISSUE #39: mwc-button: Allow overriding the disabled colors
Consider the following CSS excerpts from mwc-button-css.js.  The primary color can been customized via a CSS property.  However, the disabled state is hardcoded.  In order to make this work for my current project, I'm having to not use the built-in disabled state and instead create my own workaround.  I'm requesting that CSS properties be offered to cover the disabled state.  My particular issue is with ``mwc-button``, but I'm assuming that this will impact multiple / all components.  Thank you!

https://github.com/material-components/material-components-web-components/blob/master/packages/button/mwc-button-css.js

```css
.mdc-button:not(:disabled){color:#6200ee;color:var(--mdc-theme-primary, #6200ee)}

.mdc-button:disabled{background-color:transparent;color:rgba(0,0,0,0.37);cursor:default;pointer-events:none}
```

### PR DESCRIPTION
**Pull request by e111077**
Fixes #39 

added the following:

| Name | Default | Description
| ------------------------------------- | -------------------------------------------------- | ---
| `--mdc-button-text-transform`         | `uppercase`                                        | `text-transform` on the button label.
| `--mdc-button-letter-spacing`         | `normal`                                           | `letter-spacing` on the button label.
| `--mdc-button-horizontal-padding`     | filled: `16px` outlined: `15px` default: `8px`     | `padding-left` and `padding-right` of the button label equal to `--mdc-button-horizontal-padding` on filled buttons and `--mdc-button-horizontal-padding - --mdc-outline-width` on outlined buttons.
| `--mdc-button-outline-width`          | `1px`                                              | `width` of the outline of an outlined button and attempts to keep the component size constant.
| `--mdc-button-outline-color`          |  `--mdc-theme-primary` | Color of the outline of an outlined element. (Overrides `--mdc-theme-primary`)
| `--mdc-button-disabled-fill-color`    |  `rgba(0,0,0,0.12)`  | Background fill color of a disabled raised or unelevated button.
| `--mdc-button-disabled-ink-color`     |  `rgba(0,0,0,0.37)`  | Text color of a disabled button as well as the outline color of a disabled outlined button.
| `--mdc-button-disabled-outline-color` |  `--mdc-button-disabled-ink-color` | Sets the color of the outline of a disabled outlined button. (Overrides `--mdc-button-disabled-ink-color`)

This is what the demo looks like:

image

### CODE REVIEW

File: packages/button/README.md [hunk 1]
```diff
@@ -128,10 +128,18 @@ mwc-button {
...
 | Name | Default | Description
-| ---- | ------- | -----------
-| `--mdc-icon-font` | [`Material Icons`](https://google.github.io/material-design-icons/) | Font to use for the icon.
-| `--mdc-theme-primary` | ![](images/color_6200ee.png) `#6200ee` | Background color of the button
-| `--mdc-theme-on-primary` | ![](images/color_ffffff.png) `#ffffff` | Text color of the button
+| ------------------------------------- | -------------------------------------------------- | ---
+| `--mdc-icon-font` | [`Material Icons`](https://google.github.io/material-design-icons/)    | Font to use for the icon.
+| `--mdc-theme-primary`                 | ![](images/color_6200ee.png) `#6200ee`             | Background color of the button
+| `--mdc-theme-on-primary`              | ![](images/color_ffffff.png) `#ffffff`             | Text color of the button
+| `--mdc-button-text-transform`         | `uppercase`                                        | Sets the text-transform on the button label.
```
Comment (aomarks) [inline packages/button/README.md]: I suggest removing all the "Sets the" prefixes, they aren't really necessary.

File: packages/button/src/mwc-button.scss [hunk 2]
```diff
@@ -33,4 +49,45 @@ limitations under the License.
...
+    }
+
+    &.mdc-button--outlined {
+      border-color: var(--mdc-button-disabled-ink-color, #{$mdc-button-disabled-ink-color});
+      border-color: var(--mdc-button-disabled-outline-color, var(--mdc-button-disabled-ink-color, #{$mdc-button-disabled-ink-color}));
+    }
+  }
+}
+
+/*
```
Comment (aomarks) [inline packages/button/src/mwc-button.scss]: Remove this comment block?

File: packages/button/README.md (hunk 1 above)
Comment (e111077) [reply packages/button/README.md]: Updated description to the same text for convenience.

**Comment by aomarks** [comment]
Oops, forgot CHANGELOG -- might adding it in another one?

**Pull request merged**